modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -338,8 +342,15 @@ protected CursorStreamProviderFactory getCursorStreamProviderFactory() {
     return empty();
   }
 
+  /**
+   * Returns a (if any) {@link ConfigurationModel} that is suitable for the {@link ComponentModel}. If there are more than one
+   * {@link ConfigurationModel} that can be used by this component, the ones that can be used implicitly (according to
+   * {@link ExtensionModelUtils#canBeUsedImplicitly(ParameterizedModel)}) will be prioritized.
+   */
   private Optional<ConfigurationModel> computeConfig() {
-    return getConfigurationForComponent(extensionModel, componentModel);
+    return getConfigurationForComponent(extensionModel, componentModel).stream().sorted(reverseOrder((x, y) -> Boolean
```

Comment:
I don't agree with "then you pick the first implicit available", if there is more than one implicit we should fail. 
Selecting one implicit over another is a different story, the one where the extension developer declares a Config as a default if more than one option is available. Otherwise we are depending on how we load the configs in the extension model for selecting a config, implying that we can change the behaviour/break an application when any change in that order changes.